R/MainBar.R

```diff
@@ -8,33 +6,33 @@ Counter <- function(data, num_sets, start_col, name_of_sets, nintersections, mba
   Freqs <- data.frame()
   end_col <- as.numeric(((start_col + num_sets) -1))
   #gets indices of columns containing sets used
-  for( i in 1:num_sets){
+  for( i in seq_len(num_sets)){
     temp_data[i] <- match(name_of_sets[i], colnames(data))
   }
-  Freqs <- data.frame(count(data[ ,as.integer(temp_data)]))
-  colnames(Freqs)[1:num_sets] <- name_of_sets
+  Freqs <- plyr_count(data[ ,as.integer(temp_data)])
+  colnames(Freqs)[seq_len(num_sets)] <- name_of_sets
   #Adds on empty intersections if option is selected
-  if(is.null(empty_intersects) == F){
+  if(is.null(empty_intersects) == FALSE){
     empty <- rep(list(c(0,1)), times = num_sets)
     empty <- data.frame(expand.grid(empty))
     colnames(empty) <- name_of_sets
     empty$freq <- 0
     all <- rbind(Freqs, empty)
-    Freqs <- data.frame(all[!duplicated(all[1:num_sets]), ], check.names = F)
+    Freqs <- data.frame(all[!duplicated(all[seq_len(num_sets)]), ], check.names = FALSE)
   }
   #Remove universal empty set
-  Freqs <- Freqs[!(rowSums(Freqs[ ,1:num_sets]) == 0), ]
+  Freqs <- Freqs[!(rowSums(Freqs[ ,seq_len(num_sets)]) == 0), ]
   #Aggregation by degree
   if(tolower(aggregate) == "degree"){
-    for(i in 1:nrow(Freqs)){
-      Freqs$degree[i] <- rowSums(Freqs[ i ,1:num_sets])
+    for(i in seq_len(nrow(Freqs))){
+      Freqs$degree[i] <- rowSums(Freqs[ i ,seq_len(num_sets)])
```

Comment:
I think you can remove the `for` loop altogether. It shouldn't be necessary to iterate through the rows when using `rowSums()`.

I made similar changes back in 2021 in https://github.com/hms-dbmi/UpSetR/pull/199 to fix some computational bottlenecks that were causing us problems.

Unfortunately this repository hasn't been updated since 2020 (https://github.com/hms-dbmi/UpSetR/commit/b14854a1b066320afa913ec5560049143374a5bd), so I don't have any expectation that any of these PRs will ever be merged.